src/maximization.jl

```diff
@@ -311,6 +311,14 @@ function MAP_marg(
             diffϕ=sum(unbatch(norm(LowPass(1000) * (sqrt(ds.Cϕ) \ (ϕ - lastϕ))) / sqrt(2length(ϕ))))
         end   
         
+        push!(history, select((;g,ϕ,lastHg=Map(lastHg),diffϕ), history_keys))
```

Comment:
@marius311: For some unknown reason `Map(lastHg)` is needed for the output lastHg to be scaled correctly when hess_method="lbfgs-hessian." Otherwise, it is orders of magnitude off (and looks like it's coming from a scale factor) when passed to `history`. It has the correct amplitude when applied to ϕ in the code.

Reply:
Hmm I don't really remember tbh but glancing at https://github.com/marius311/CMBLensing.jl/blob/ec4bd106de5067c811ed6a65c96ca7c2b26ab8af/src/maximization.jl#L200-L212 looks like I also have some `Map`s. I think in theory you could get rid of that by defining more of the things OptimKit needs like `retract`, `inner` (that one you already are), `scale!`, `add!`, and `transport!` as mentioned in their readme, although my guess is performance-wise the extra `Map` don't really matter so its probably fine.

Reply:
I did see that you pass in `Map`-converted field variables in `MAP_joint` for the optimization in places. I tested both and in my case, they yield the same results with or without passing a `Map`-converted field and figure to just go without to reduce the back-and-forth.

Agreed that it doesn't slow down the code. But it does mean it's storing a larger vector (Map vs Fourier), and it's not the same (Fourier)type as the rest of the return keys. So I want to see if you already know of similar peculiar behavior. Or if this is a corner case, that I run into.